salt/modules/aptly.py

```diff
@@ -0,0 +1,517 @@
+    :param str distribution: The default distribution to use when publishing.
+    :param str uploaders_file: The repository upload restrictions config.
+    :param str saltenv: The environment the file resides in.
+
+    :return: A list of the arguments formatted as aptly arguments.
+    :rtype: list
+    '''
+    ret = list()
+    cached_uploaders_path = None
+    settings = {'comment': comment, 'component': component,
+                'distribution': distribution}
+
+    if uploaders_file:
+        cached_uploaders_path = __salt__['cp.cache_file'](uploaders_file, saltenv)
+
+        if not cached_uploaders_path:
+            _LOG.error('Unable to get cached copy of file: %s', uploaders_file)
+            return False
+
+    for setting in settings:
+        if settings[setting] is not None:
+            ret.append('-{}={}'.format(setting, settings[setting]))
+
+    if cached_uploaders_path:
+        ret.append('-uploaders-file={}'.format(cached_uploaders_path))
+
+    return ret
+
+
+def _validate_config(config_path):
+    '''
+    Validate that the configuration file exists and is readable.
+
+    :param str config_path: The path to the configuration file for the aptly instance.
+
+    :return: None
+    :rtype: None
+    '''
+    _LOG.debug('Checking configuration file: %s', config_path)
+
+    if not os.path.isfile(config_path):
+        message = 'Unable to get configuration file: {}'.format(config_path)
+        _LOG.error(message)
+        raise SaltInvocationError(message)
+
+
+def get_config(config_path=_DEFAULT_CONFIG_PATH):
+    '''
+    Get the configuration data.
+
+    :param str config_path: The path to the configuration file for the aptly instance.
+
+    :return: A dictionary containing the configuration data.
+    :rtype: dict
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' aptly.get_config
+    '''
+    _validate_config(config_path)
+
+    cmd = ['config', 'show', '-config={}'.format(config_path)]
+
+    cmd_ret = _cmd_run(cmd)
+
+    return json.loads(cmd_ret)
+
+
+def list_repos(config_path=_DEFAULT_CONFIG_PATH, with_packages=False):
+    '''
+    List all of the repos.
+
+    :param str config_path: The path to the configuration file for the aptly instance.
+    :param bool with_packages: Return a list of packages in the repo.
+
+    :return: A dictionary of the repositories.
+    :rtype: dict
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' aptly.list_repos
+    '''
+    _validate_config(config_path)
+
+    ret = dict()
+    cmd = ['repo', 'list', '-config={}'.format(config_path), '-raw=true']
+
+    cmd_ret = _cmd_run(cmd)
+    repos = [line.strip() for line in cmd_ret.splitlines()]
+
+    _LOG.debug('Found repositories: %s', len(repos))
+
+    for name in repos:
+        ret[name] = get_repo(name=name, config_path=config_path,
+                             with_packages=with_packages)
+    return ret
+
+
+def get_repo(name, config_path=_DEFAULT_CONFIG_PATH, with_packages=False):
```

Comment:
Maybe `return_packages` would be a better name? Not sure.

Reply:
I was trying to stay as close to the aptly command line argument names as possible. Plus, `return_packages` might give the impression that it returns only the packages in the repo and not other metadata.

Reply:
Sounds good. Thanks!